usr/share/rear/lib/bareos-functions.sh

```diff
@@ -0,0 +1,301 @@
+#
+# Bareos helper functions.
+#
+
+#
+# helper functions for easier handling bconsole output
+# (remove irrelevant parts of the output).
+#
+function bcommand()
+{
+  local out
+  out=$(mktemp)
+  (
+    for i in "${BCOMMAND_PRE_COMMANDS[@]}"; do
+        echo "$i"
+    done
+    echo "@tee $out"
+    for i in "$@"; do
+        echo "$i"
+    done
+  ) | bconsole > /tmp/bconsole.$!
+  rc=$?
+  # BCOMMAND_PRE_COMMANDS have been executed
+  # and are therefore unset.
+  unset BCOMMAND_PRE_COMMANDS
+
+  # remove submitted commands from output.
+  local sed_args="(You have messages."
+  for i in "$@"; do
+    sed_args+="|$i"
+  done
+  sed_args+=")"
+
+  sed -r -e "/^${sed_args}$/d" -e "s/${sed_args}$//" < "$out"
+  rm "$out"
+  return $rc
+}
+
+function bcommand_json()
+{
+  BCOMMAND_PRE_COMMANDS=( ".api json compact=no" )
+  bcommand "$@"
+  unset BCOMMAND_PRE_COMMANDS
+  return $?
+}
+
+function bcommand_extract_value()
+{
+  local key="$1"
+  local sed_arg
+  sed_arg="$(printf 's/^ *%s: (.*) *$/\\1/p' "$key")"
+  sed -n -r "${sed_arg}"
+}
+
+function bcommand_check_client_status()
```

Comment:
Please rename the function to something that indicates that it doesn't just return an info but actually errors out. Like `bareos_ensure_usable_client_status` or such. Then you won't have to comment that this function errors out

Reply:
Makes sense.

Reply:
Let's keep it open till after you change it, otherwise I'll have a hard time keeping track